network/src/protocols/health_checker/test.rs

```diff
@@ -155,205 +167,138 @@ async fn expect_pong(res_rx: oneshot::Receiver<Result<Bytes, RpcError>>) {
-        send_new_peer_notification(peer_id, &mut connection_notifs_tx).await;
+        harness.send_new_peer_notification(peer_id).await;
 
         // Receive ping from peer.
-        let res_rx = send_inbound_ping(peer_id, 0, &mut network_notifs_tx).await;
+        let res_rx = harness.send_inbound_ping(peer_id, 0).await;
 
         // HealthChecker should respond with a pong.
         expect_pong(res_rx).await;
     };
-    rt.block_on(events_f);
+    block_on(future::join(health_checker.start(), test));
 }
 
 #[test]
 fn outbound_failure_permissive() {
-    ::diem_logger::Logger::init_for_testing();
-    let mut rt = Runtime::new().unwrap();
     let ping_failures_tolerated = 10;
-    let (mut network_reqs_rx, _, mut connection_reqs_rx, mut connection_notifs_tx, mut ticker_tx) =
-        setup_permissive_health_checker(&mut rt, ping_failures_tolerated);
+    let (mut harness, health_checker) = TestHarness::new_permissive(ping_failures_tolerated);
 
-    let events_f = async move {
+    let test = async move {
         // Trigger ping to a peer. This should do nothing.
-        ticker_tx.send(()).await.unwrap();
+        harness.trigger_ping().await;
 
         // Notify HealthChecker of new connected node.
         let peer_id = PeerId::random();
-        send_new_peer_notification(peer_id, &mut connection_notifs_tx).await;
+        harness.send_new_peer_notification(peer_id).await;
 
         // Trigger pings to a peer. These should ping the newly added peer, but not disconnect from
         // it.
         for _ in 0..=ping_failures_tolerated {
-            ticker_tx.send(()).await.unwrap();
             // Health checker should send a ping request which fails.
-            expect_ping_send_notok(&mut network_reqs_rx).await;
+            harness.trigger_ping().await;
+            harness.expect_ping_send_notok().await;
         }
 
         // Health checker should disconnect from peer after tolerated number of failures
-        expect_disconnect(peer_id, &mut connection_reqs_rx).await;
+        harness.expect_disconnect(peer_id).await;
     };
-    rt.block_on(events_f);
+    block_on(future::join(health_checker.start(), test));
 }
 
 #[test]
 fn ping_success_resets_fail_counter() {
-    ::diem_logger::Logger::init_for_testing();
-    let mut rt = Runtime::new().unwrap();
     let failures_triggered = 10;
     let ping_failures_tolerated = 2 * 10;
-    let (mut network_reqs_rx, _, mut connection_reqs_rx, mut connection_notifs_tx, mut ticker_tx) =
-        setup_permissive_health_checker(&mut rt, ping_failures_tolerated);
+    let (mut harness, health_checker) = TestHarness::new_permissive(ping_failures_tolerated);
 
-    let events_f = async move {
+    let test = async move {
         // Trigger ping to a peer. This should do nothing.
-        ticker_tx.send(()).await.unwrap();
+        harness.trigger_ping().await;
+
         // Notify HealthChecker of new connected node.
         let peer_id = PeerId::random();
```

Comment:
Nit: For any of these where we're calling from the mocked peer, to a different peer, we should hardcode the second `PeerId`.  Otherwise, with the random it adds variability, which doesn't really lead any insight, except some **really small** chance that the test fails.